parseMMCIF fix for atomnames with prime

jamesmkrieger opened the pull request:
Without this, the double quotes around atom names with primes in mmCIF files are included in the atom names:
```
Python 3.9.9 | packaged by conda-forge | (main, Dec 20 2021, 02:40:17) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.31.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from prody import *

In [2]: ag = parseMMCIF('4v8r')
@> 128780 atoms and 1 coordinate set(s) were parsed in 2.50s.

In [3]: ag['VD'].getNames()
Out[3]: 
array(['PB', 'O1B', 'O2B', 'O3B', 'PA', 'O1A', 'O2A', 'O3A', '"O5\'"',
       '"C5\'"', '"C4\'"', '"O4\'"', '"C3\'"', '"O3\'"', '"C2\'"',
       '"O2\'"', '"C1\'"', 'N9', 'C8', 'N7', 'C5', 'C6', 'N6', 'N1', 'C2',
       'N3', 'C4'], dtype='<U6')
```

These then get transferred when writing to PDB and make the lines longer:
```
HETATM1f6f4  O3A ADP Bz4601     180.693 -58.120 -21.066  1.00148.50        VD O  
HETATM1f6f5 "O5'" ADP Bz4601     182.720 -58.776 -22.315  1.00146.40        VD O 
```

With the fix, this no longer happens:
```
In [3]: ag['VD'].getNames()
Out[3]: 
array(['PB', 'O1B', 'O2B', 'O3B', 'PA', 'O1A', 'O2A', 'O3A', "O5'", "C5'",
       "C4'", "O4'", "C3'", "O3'", "C2'", "O2'", "C1'", 'N9', 'C8', 'N7',
       'C5', 'C6', 'N6', 'N1', 'C2', 'N3', 'C4'], dtype='<U6')

In [4]: writePDB('4v8r_fix1', ag)
@> WARNING Indices are exceeding 99999 and hexadecimal format is being used for indices
Out[4]: '4v8r_fix1.pdb'
```

```
HETATM1f6f4  O3A ADP Bz4601     180.693 -58.120 -21.066  1.00148.50        VD O  
HETATM1f6f5  O5' ADP Bz4601     182.720 -58.776 -22.315  1.00146.40        VD O  
```

This is in line with the original PDB file 4aol, which contained part of the structure:
```
HETATM63902  O3A ADP A1551     102.408  -0.305 160.824  1.00 92.70           O  
HETATM63903  O5' ADP A1551     102.826  -1.114 163.128  1.00141.40           O 
```

```
In [5]: aol = parsePDB('4aol', compressed=False)
@> PDB file is found in working directory (4aol.pdb).
@> 64390 atoms and 1 coordinate set(s) were parsed in 0.78s.
@> Secondary structures were assigned to 5486 residues.

In [6]: list(aol.hetatm.getHierView())[0].getNames()
Out[6]: 
array(['PB', 'O1B', 'O2B', 'O3B', 'PA', 'O1A', 'O2A', 'O3A', "O5'", "C5'",
       "C4'", "O4'", "C3'", "O3'", "C2'", "O2'", "C1'", 'N9', 'C8', 'N7',
       'C5', 'C6', 'N6', 'N1', 'C2', 'N3', 'C4', 'BE', 'F1', 'F2', 'F3',
       'MG'], dtype='<U6')
```

jamesmkrieger commented:
Checks passed: https://github.com/jamesmkrieger/ProDy/actions/runs/1852947326

This is unsurprising as there are no checks related to this very localised error

SHZ66 commented on prody/proteins/ciffile.py:
```diff
@@ -290,6 +290,8 @@ def _parseMMCIFLines(atomgroup, lines, model, chain, subset,
 
         atomname = line.split()[fields['auth_atom_id']]
+        if atomname.find('"') != -1:
+            atomname = atomname[1:4]
```
The fix looks fine to me. But Since any quotes in the atom name will trigger this, maybe the fix can be more aggressive too?
```
atomname = atomname.replace('"', '')
```

jamesmkrieger replied:
> The fix looks good and should make it to the release. However, I wonder how the quotes get into the atom's name in the first place.

They're in the mmCIF file:
```
In [3]: fi = open('/mnt/c/Users/james/OneDrive/Documents/MC_fellowship/4_other_systems/1_CCT/4v8r.cif', 'r')

In [4]: lines = fi.readlines()

In [5]: fi.close()

In [6]: line = lines[137924]

In [7]: line
Out[7]: 'HETATM 128757 O  "O5\'" . ADP VD 9  .   ? 182.720 -58.776 -22.315 1.00 146.40 ? 4601 ADP Bz "O5\'" 1 \n'

In [8]: line.split()
Out[8]: 
['HETATM',
 '128757',
 'O',
 '"O5\'"',
 '.',
 'ADP',
 'VD',
 '9',
 '.',
 '?',
 '182.720',
 '-58.776',
 '-22.315',
 '1.00',
 '146.40',
 '?',
 '4601',
 'ADP',
 'Bz',
 '"O5\'"',
 '1']

In [9]: line.split()[3]
Out[9]: '"O5\'"'
```

jamesmkrieger replied:
Maybe I should be left aggressive actually instead. It could be that there are some we want.

jamesmkrieger replied:
How about this?
```
        if atomname.startswith('"') and atomname.endswith('"'):
            atomname = atomname[1:-1]
```

SHZ66 commented:
How did the quotes get into the atom's name in the first place?

jamesmkrieger commented:
> The fix looks good and should make it to the release. However, I wonder how the quotes get into the atom's name in the first place.

They're in the mmCIF file:
```
In [3]: fi = open('/mnt/c/Users/james/OneDrive/Documents/MC_fellowship/4_other_systems/1_CCT/4v8r.cif', 'r')

In [4]: lines = fi.readlines()

In [5]: fi.close()

In [6]: line = lines[137924]

In [7]: line
Out[7]: 'HETATM 128757 O  "O5\'" . ADP VD 9  .   ? 182.720 -58.776 -22.315 1.00 146.40 ? 4601 ADP Bz "O5\'" 1 \n'

In [8]: line.split()
Out[8]: 
['HETATM',
 '128757',
 'O',
 '"O5\'"',
 '.',
 'ADP',
 'VD',
 '9',
 '.',
 '?',
 '182.720',
 '-58.776',
 '-22.315',
 '1.00',
 '146.40',
 '?',
 '4601',
 'ADP',
 'Bz',
 '"O5\'"',
 '1']

In [9]: line.split()[3]
Out[9]: '"O5\'"'
```
